### PR TITLE
chore: rename 'master' to 'main' in docs and tooling

### LINKED ISSUE #5051: 🛠 Repo: Rename master branch to main
### Tooling Suggestion Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `master` branch of the repository.
- [X] I have read and agree to Mocha's [Code of Conduct](https://github.com/mochajs/mocha/blob/master/.github/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md)
- [X] I have searched for [related issues](https://github.com/mochajs/mocha/issues?q=is%3Aissue) and [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20), but none matched my issue.
- [X] I want to provide a PR to resolve this

### Overview

Per https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main, the default branch for repositories is now `main`. We're going to apply the same change to this repo and its tooling. Filing this issue for tracking.

### Additional Info

https://github.com/github/renaming has more details on the context and practices around the rename.

This'll need a find-and-replace in any docs, workflows, etc. that refer to branches/Git/etc.

Intentional follow-up to #5038.

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

- [x] Addresses an existing open issue: fixes #5051
- [x] That issue was marked as [`status: accepting prs`](https://github.com/mochajs/mocha/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/mochajs/mocha/blob/master/.github/CONTRIBUTING.md) were taken

## Overview

Roughly a straightforward find-and-replace.

Once this is approved, I can rename the branches here and in github.com/mochajs/mocha-examples.

### CODE REVIEW

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -100,4 +100,4 @@ This is a foundation-wide team established to manage escalation when a reporter
 In order to escalate to the CoCP send an email to `"coc-escalation@lists.openjsf.org`.
 
 For more information, refer to the full
-[Code of Conduct governance document](https://github.com/openjs-foundation/cross-project-council/blob/master/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md).
+[Code of Conduct governance document](https://github.com/openjs-foundation/cross-project-council/blob/main/FOUNDATION_CODE_OF_CONDUCT_REQUIREMENTS.md).
```
Comment (JoshuaKGoldberg) [inline .github/CODE_OF_CONDUCT.md:103]: This is an existing 404. I think https://github.com/openjs-foundation/cross-project-council/tree/main/proposals/approved/CODE_OF_CONDUCT is the replacement? Not sure.

```suggestion
[Code of Conduct governance document](https://github.com/openjs-foundation/cross-project-council/tree/main/proposals/approved/CODE_OF_CONDUCT).
```

File: docs/index.md [hunk 2]
```diff
@@ -2362,14 +2362,14 @@ or the [source](https://github.com/mochajs/mocha/blob/master/lib/mocha.js).
 [discord-mocha]: https://discord.gg/KeDn2uXhER
 [emacs]: https://www.gnu.org/software/emacs/
 [emacs-mocha.el]: https://github.com/scottaj/mocha.el
-[example-babel]: https://github.com/mochajs/mocha-examples/tree/master/packages/babel
+[example-babel]: https://github.com/mochajs/mocha-examples/tree/main/packages/babel
 [example-connect-test]: https://github.com/senchalabs/connect/tree/master/test
```
Comment (akashchouhan16) [inline docs/index.md:2366]: Looks like you missed changing this  `main`  for this one *line_2366* & _line_2367_

File: docs/index.md [hunk 3]
```diff
@@ -2362,14 +2362,14 @@ or the [source](https://github.com/mochajs/mocha/blob/master/lib/mocha.js).
...
-[example-mocha-test]: https://github.com/mochajs/mocha/tree/master/test
-[example-mocha-config]: https://github.com/mochajs/mocha/tree/master/example/config
+[example-mocha-test]: https://github.com/mochajs/mocha/tree/main/test
+[example-mocha-config]: https://github.com/mochajs/mocha/tree/main/example/config
 [example-superagent-test]: https://github.com/visionmedia/superagent/tree/master/test/node
-[example-third-party-reporter]: https://github.com/mochajs/mocha-examples/tree/master/packages/third-party-reporter
-[example-typescript]: https://github.com/mochajs/mocha-examples/tree/master/packages/typescript
+[example-third-party-reporter]: https://github.com/mochajs/mocha-examples/tree/main/packages/third-party-reporter
+[example-typescript]: https://github.com/mochajs/mocha-examples/tree/main/packages/typescript
 [example-websocket.io-test]: https://github.com/LearnBoost/websocket.io/tree/master/test
```
Comment (akashchouhan16) [inline docs/index.md:2373]: same here, missed url change to `main`

File: docs/index.md (hunk 2 above)
Comment (JoshuaKGoldberg) [reply docs/index.md:2366]: Thanks for the spot! That's an external repo that still uses `master`, not us. 🙂

**Comment by voxpelli** [comment]
lgtm

**Pull request merged**